src/test/java/TestCalculator.java

```diff
@@ -0,0 +1,137 @@
+import org.junit.After;
+import org.testng.Assert;
+import org.testng.annotations.*;
+import org.testng.asserts.SoftAssert;
+
+public class TestCalculator  { // Do not extend it from parent Calculator class
+
+//    Calculator calc = new Calculator(); // Not really effective implementation, better to instantiate object in @BeforeMethod
+    private Calculator calc; // Declaring variable of Calculator class
+
+    @DataProvider(name = "myDataProvider")
+    public static Object[][] getData(){
+        return new Object[][]{
+                {14, 7},
+                {8, 0}
+        };
+    }
+
+    @BeforeSuite
+    public void beforeSuiteMethod(){
+        System.out.println("This is method executed before running Test Suite");
+    }
+
+    @BeforeGroups
+    public void beforeGroupsMethod(){
+        System.out.println("This is method executed before running Groups");
+    }
+
+    @BeforeTest
+    public void beforeTestMethod(){
+        System.out.println("This is method executed before running Test");
+    }
+
+    @BeforeClass
+    public void beforeClassMethod(){
+        System.out.println("This is method executed before running Class");
+    }
+
+    @BeforeMethod (groups = "failedgroup") // If to remove "(groups = "failedgroup")" from here, all tests from testng2.xml will fail
+    public void setUp(){
+        calc = new Calculator();
+        System.out.println("This is method executed before running Method");
+    }
+
+    @AfterSuite
+    public void afterSuiteMethod(){
+        System.out.println("This is method executed after running Test Suite");
+    }
+
+    @AfterGroups
+    public void afterGroupsMethod(){
+        System.out.println("This is method executed after running Groups");
+    }
+
+    @AfterTest
+    public void afterTestMethod(){
+        System.out.println("This is method executed after running Test");
+    }
+
+    @AfterClass
+    public void afterClassMethod(){
+        System.out.println("This is method executed after running Class");
+    }
+
+    @AfterMethod
+    public void tearDown(){
+        System.out.println("This is method executed after running Method");
+    }
+
+    @Test (groups = "passedgroup", priority = 1) //Test with parameters from testng.xml
+    @Parameters ({"a", "b"})
+    public void testAdd(int x, int y){
```

Comment:
For parameters from suite, it is recommended to add default value with @Optional
In this case you will be able to run test straight from the test file